src/operators.jl

```diff
@@ -89,58 +76,42 @@ end
 # Note this list does not include ^, ÷ and %
 # Operations between signed and unsigned types are not safe: promotion to unsigned
 # gives an InexactError for negative numbers
-if VERSION >= v"0.5.0-dev"
-    for op in (:+, :-, :*, :/, :&, :|, :<<, :>>, :(>>>),
-               :(==), :<, :>, :<=, :>=,
-               :scalarmin, :scalarmax,
-               :isless)
-        @eval begin
-            # to fix ambiguities
-            null_safe_op{S<:SafeFloats,
-                         T<:SafeFloats}(::typeof($op), ::Type{S}, ::Type{T}) = true
-            null_safe_op{S<:SafeSigned,
-                         T<:SafeSigned}(::typeof($op), ::Type{S}, ::Type{T}) = true
-            null_safe_op{S<:SafeUnsigned,
-                         T<:SafeUnsigned}(::typeof($op), ::Type{S}, ::Type{T}) = true
-        end
-    end
-else # No functors for all methods on 0.4: use the slow path for missing ones
-    for op in (:+, :-, :*, :/, :&, :|,
-               :<, :>,
-               :scalarmin, :scalarmax)
-        @eval begin
-            # to fix ambiguities
-            null_safe_op{S<:SafeFloats,
-                         T<:SafeFloats}(::typeof($op), ::Type{S}, ::Type{T}) = true
-            null_safe_op{S<:SafeSigned,
-                         T<:SafeSigned}(::typeof($op), ::Type{S}, ::Type{T}) = true
-            null_safe_op{S<:SafeUnsigned,
-                         T<:SafeUnsigned}(::typeof($op), ::Type{S}, ::Type{T}) = true
-        end
+for op in (:+, :-, :*, :/, :&, :|, :<<, :>>, :(>>>),
+           :(==), :<, :>, :<=, :>=,
+           :scalarmin, :scalarmax,
+           :isless)
+    @eval begin
+        # to fix ambiguities
+        null_safe_op{S<:SafeFloats,
```

Comment:
Actually I think I prefer it this way, since it shows that the explicit prefix implies extension and no prefix implies using the function. But if you feel strongly I can change it.